### PR TITLE
Refactor profiles; add locally approximated profiles

### PR DESCRIPTION
**Pull request by yannikschaelte**
Regarding optimization based profiles:
* Fix some errors when having multiple profile lists
* Fix (silently passing) tests
* Correct docstrings
* Rename some inner functions to better describe their action

Introduce local approximation based profiles:
* Add `pypesto.profile.approximate_parameter_profiles` function
* Add `pypesto.visualize.profile_cis` visualization.
* Illustrate in rosenbrock nb.

### CODE REVIEW

File: pypesto/profile/util.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    -------
+    ratio:
+        Corresponds to a likelihood ratio.
+    """
+    quantile = scipy.stats.chi2.ppf(alpha, df=1)
+    ratio = np.exp(-quantile / 2)
+    return ratio
+
+
+def initialize_profile(
```
Comment (yannikschaelte) [inline pypesto/profile/util.py:33]: note: the below functions are basically copy-pasted (with better docs)

File: pypesto/profile/approximate.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    x = optimizer_result.x
+    fval = optimizer_result.fval
+    hess = problem.get_reduced_matrix(optimizer_result.hess)
+
+    # ratio scaling factor
+    ratio_scaling = np.exp(global_opt - fval)
+
+    # we need the hessian - compute if not provided or fishy
+    if hess is None or np.isnan(hess).any():
+        hess = problem.objective(
```
Comment (FFroehlich) [inline pypesto/profile/approximate.py:78]: log.info/warning whats going on, may be computationally expensive to compute

File: pypesto/visualize/profile_cis.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    # calculate intervals
+    intervals = []
+    for i_par in range(problem.dim_full):
+        if i_par not in profile_indices:
+            continue
+        xs = profile_list[i_par].x_path[i_par]
+        ratios = profile_list[i_par].ratio_path
+        indices, = np.where(ratios > confidence_ratio)
+        lb, ub = xs[indices[0]], xs[indices[-1]]
```
Comment (FFroehlich) [inline pypesto/visualize/profile_cis.py:64]: why not estimate values based on linear interpolation?

Comment (yannikschaelte) [reply pypesto/visualize/profile_cis.py]: agreed, should in particular be done for coarser profile steps.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/approximate.py  7
- pypesto/visualize/profiles.py  3
- test/test_profile.py  2
- pypesto/visualize/profile_cis.py  10
- test/test_problem.py  4
- pypesto/profile/util.py  9
         

Complexity decreasing per file
==============================
+ pypesto/profile/profile.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5613274&bid=18388229)

**Pull request merged**